### DISCUSSION TITLE
How to invoke Formatter from CLI

### DISCUSSION THREAD

**Discussion by jiashengguo** [eclipse-langium/langium#1305] [Q&A] [answered]
I have already implemented my AbstractFormatter in the language server.  How to invoke it from the CLI?

**Comment by msujew** [comment, marked as answer]
@jiashengguo You can invoke the formatter programmatically and apply the changes that it generates to the specified document. See [our formatter test functions](https://github.com/eclipse-langium/langium/blob/750ff44c4d4cf3edee108b2998e818525a2b9ecc/packages/langium/src/test/langium-test.ts#L472-L483) on how to do that.

**Reply by jiashengguo** [reply]
Thanks!
